### PR TITLE
Fix invalid uint slices

### PR DESCRIPTION
**Pull request by janderland**

### CODE REVIEW

File: orderedmap_test.go [hunk 1]
```diff
@@ -530,3 +532,24 @@ func TestOrderedMap_empty_map(t *testing.T) {
...
+
+	str2, err := json.Marshal(om)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(str1, str2) {
+		t.Fatalf("%s != %s", str1, str2)
+	}
+}
```
Comment (stetra) [inline orderedmap_test.go]: nit: missing newline

File: orderedmap.go [hunk 2]
```diff
@@ -3,19 +3,19 @@ package orderedmap
 import (
```
Comment (stetra) [inline orderedmap.go:3]: There are still unchecked `json.Unmarshal` calls such as on [this line](https://github.com/Pexeso/orderedmap/blob/00f886ea4c249aff10285c690e6ba4e73ccb2c8c/orderedmap.go#L177). Should these also be updated to be absolutely sure that no edge cases are missed?

Comment (janderland) [reply orderedmap.go:3]: A lot of the `json.Unmarshal` calls are used to just check for valid JSON syntax, and their output values aren't even used, so I left those around. I could get rid of all the `json.Unmarshal` and the `setNumber` option completely and just always have the effects of `setNumber == true`. Would you prefer that?

File: orderedmap_test.go (hunk 1 above)
Comment (janderland) [reply orderedmap_test.go]: Fixed.

File: orderedmap.go (hunk 2 above)
Comment (stetra) [reply orderedmap.go:3]: Ah that's fine then. LGTM

Comment (janderland) [reply orderedmap.go:3]: Done.

**Pull request merged**